src/exporter/DaemonMetricCollector.cc

```diff
@@ -147,23 +147,24 @@ void DaemonMetricCollector::dump_asok_metrics() {
           std::string counter_name = perf_group + "_" + counter_name_init;
           promethize(counter_name);
 
-          if (counters_labels.empty()) {
-            auto labels_and_name = get_labels_and_metric_name(daemon_name, counter_name);
-            if (labels_and_name.first.empty()) {
-              dout(1) << "Unable to parse instance_id from daemon_name: " << daemon_name << dendl;
-              continue;
-            }
-            labels = labels_and_name.first;
-            counter_name = labels_and_name.second;
+          auto labels_and_name = get_labels_and_metric_name(daemon_name, counter_name);
+          if (labels_and_name.first.empty()) {
+            dout(1) << "Unable to parse instance_id from daemon_name: " << daemon_name << dendl;
+            continue;
           }
+          labels.insert(labels_and_name.first.begin(), labels_and_name.first.end());
+          counter_name = labels_and_name.second;
+
           // For now this is only required for rgw multi-site metrics
           auto multisite_labels_and_name = add_fixed_name_metrics(counter_name);
           if (!multisite_labels_and_name.first.empty()) {
             labels.insert(multisite_labels_and_name.first.begin(), multisite_labels_and_name.first.end());
             counter_name = multisite_labels_and_name.second;
           }
-          auto perf_values = counters_values.at(counter_name_init);
-          dump_asok_metric(counter_group, perf_values, counter_name, labels);
+          if (counters_values.find(counter_name_init) != counters_values.end()) {
```

Comment:
This change seems unrelated to the issue being addressed here -- what is this `if` condition for?

How can a counter which is present in `counters` (where `counter_name_init` comes from) be not present in `counters_values`?

Reply:
Yes it shouldn't include that change. Got those from previous cherry-picks. Removed them!